lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp

```diff
@@ -4320,6 +4320,49 @@ class DecomposeAtenVarMeanDimOp : public OpRewritePattern<AtenVarMeanDimOp> {
 };
 } // namespace
 
+namespace {
+// Decompose `aten.topk` op into `aten.sort` and `aten.slice.Tensor` op.
+class DecomposeAtenTopkOp : public OpRewritePattern<AtenTopkOp> {
+public:
+  using OpRewritePattern::OpRewritePattern;
+  LogicalResult matchAndRewrite(AtenTopkOp op,
+                                PatternRewriter &rewriter) const override {
+    Location loc = op.getLoc();
+    auto context = op.getContext();
+
+    bool sorted;
+    if (!matchPattern(op.getSorted(), m_TorchConstantBool(&sorted)))
+      return rewriter.notifyMatchFailure(
+          op, "Expected a constant boolean value for sorted");
+    if (!sorted)
+      return rewriter.notifyMatchFailure(
+          op, "unimplemented: sorted value arg must be set to True");
+
+    Value self = op.getSelf();
+    Value dim = op.getDim();
+    auto selfType = self.getType().cast<ValueTensorType>();
+    auto sortIndicesType = ValueTensorType::get(
```

Comment:
nit: I would cast to `BaseTensorType` and use `selfType.getWithSizesAndDtype` to define `sortIndicesType`, since there is no guarantee that things will have value semantics when this pattern is applied.